openlibrary/fastapi/search.py

```diff
@@ -1,52 +1,116 @@
 from __future__ import annotations
 
-from fastapi import APIRouter, Query, Request
+import json
+from typing import Annotated, Any
+
+from fastapi import APIRouter, Depends, Query, Request
 from fastapi.responses import JSONResponse
+from pydantic import BaseModel, create_model
 
-from openlibrary.plugins.worksearch.code import work_search_async
+from openlibrary.plugins.worksearch.code import (
+    default_spellcheck_count,
+    work_search_async,
+)
 from openlibrary.plugins.worksearch.schemes.works import WorkSearchScheme
 
 router = APIRouter()
 
+ListQuery = Annotated[list[str], Query(default_factory=list)]
+
+
+# Ideally this will go in a models files, we'll move it for the 2nd endpoint
+class Pagination(BaseModel):
+    limit: Annotated[int, Query(ge=0)] = 100
+    offset: Annotated[int | None, Query(ge=0)] = None
+    page: Annotated[int | None, Query(ge=1)] = None
+
+    def model_post_init(self, _):
+        if self.offset is not None:
+            self.page = None
+        else:
+            self.page = self.page or 1
+
+
+field_names = set(WorkSearchScheme.all_fields) | set(
+    WorkSearchScheme.field_name_map.keys()
+)
+
+# Dynamically create the model
+AllAllowedParams = create_model(
```

Comment:
Discussed on call ; turns out only the facet fields of the WorkSearchScheme and this set called `check_params` are what's allowed. Longer term, it would make sense to nix this list from the WorkSearchScheme and have that be the responsibility of the API. But for now copying it here makes sense.